### ISSUE TITLE
Feature Request - Other ID metadata (non-core)

### ISSUE THREAD

**Issue by dustymc** [ArctosDB/arctos#4604]
**Is your feature request related to a problem? Please describe.**
 
See https://github.com/ArctosDB/arctos/issues/4101#issuecomment-1099410385; we need to flesh out the "more data" table. 

**Describe what you're trying to accomplish** 

Add more data to other identifiers

**Additional context**

``assigned_agent_id`` and ``assigned_date`` (the who and when describing how the identifier became attached to a record) will be in the core model for next release. 

**Describe the solution you'd like**

A resolution of what's necessary. The proposed model is more agent-centric than typical "attributes." I would like to clarify the purpose and a refine the data structure before proceeding. 

Here's a slight refinement of the proposal from the other issue. 

table identifier_metadata:

|field|references|null|what?|
|--|--|--|--|
|identifier_metadata_id||no|primary key|
|coll_obj_other_id_num_id|coll_obj_other_id_num(coll_obj_other_id_num_id)|no|link to an identifier|
|agent_id|agent(agent_id)|no|agent making the assertion|
|agent_role|ctidentifier_agent_role(identifier_agent_role)|no|new code table, see below|
|date||yes|date on which the assertion was made; tentatively NULLable iso8601 date but maybe a not-null DATE object make more sense?|
|remark||yes|free text|

``ctidentifier_agent_role.identifier_agent_role`` (better names very welcome) would be a new table containing "things agents do to identifiers."

|possible value|overly-vague documentation|
|--|--|
|issued by|agent minting the identifier|
|accepted by|agent accepting the identifier|
|rejected by|agent refuting the identifier|

So, given some identifier (always accompanied by ``assigned_agent_id`` and ``assigned_date``), this could be used for...

|agent_role|agent|date|remark|
|--|--|--|--|
|issued by|MSB|{somedate}|NK page printed on {somedate}|
|accepted by|me|yesterday|this looks correct|
|rejected by|you|today|this is wrong, hopefully some elaboration on that....|
|rejected by|someone_else|tomorrow|the script that made this is busted, bla bla....|

... or ...

|agent_role|agent|date|remark|
|--|--|--|--|
|issued by|{the collector}|{somedate}|Number taken from {the collector}'s field notes, maybe some details on that....|

Is this a suitable model, are there any use cases this can't sufficiently address?

**Priority**

Suggest low until there's a clear use case with actual data.

**Comment by Jegelewicz** [comment]
I would say the priority is high - we need this if we want to let GloBI help us out with related stuff.

**Comment by dustymc** [comment]
>let GloBI help 

I personally would be happy to set them up with ``assigned_agent_id`` and ``assigned_date`` (which will be in next release) - that's enough to get at "this is from a script" and "the script changed on {date} so maybe old stuff needs flushed" and etc.

That also applies to me - if you want something beyond those two core fields for reciprocals, derivatives, things I can figure out from Organism ID, etc., then I need to know ASAP.

**Comment by campmlc** [comment]
What's in next release, the full model above or just those two fields? We certainly need remarks, I like the issued by, accepted by, rejected by etc. What would we need different for reciprocals?
Agree priority is high, so we should move forward with basics for GloBI, add easy stuff now.

**Comment by campmlc** [comment]
This issue affects so many different projects and upcoming conference
presentations and collaborations, then we need to prioritize resolution.
Can we resolve at AWG this week?

**Comment by Jegelewicz** [comment]
I am in favor of implementing this. Just one question. Will the above REPLACE what is there now or will it be an add on?

When I look at a collector number now, I see this:

![image](https://user-images.githubusercontent.com/5725767/165766808-0816a231-f1e4-45bb-b426-8e838c680815.png)

Would that then become

ID type | prefix or string| ID number | suffix | relationship | assigned by | agent role | agent | date | remark 
 -- | -- | -- | -- | -- | -- | -- | -- | -- | -- 
preparator number | ```DND&nbsp;``` | 3 | | self | legacy | issued by | [Delaine N. Dalager](https://arctos.database.museum/agent/21324779) | 1970-05-30 | from collector notes 

or 

ID type | prefix or string| ID number | suffix | relationship | agent role | agent | date | remark 
 -- | -- | -- | -- | -- | -- | -- | -- | -- 
preparator number | ```DND&nbsp;``` | 3 | | self | issued by | [Delaine N. Dalager](https://arctos.database.museum/agent/21324779) | 1970-05-30 | 
preparator number | ```DND&nbsp;``` | 3 | | self | assigned by | legacy |  | from collector notes

**Comment by dustymc** [comment]
add on

* current stuff (the usual plus "entered into Arctos by agent on date" - and not in any way replicated by the addition of "metadata," just to be clear)
   * issued by [Delaine N. Dalager](https://arctos.database.museum/agent/21324779)
   * assigned by [Delaine N. Dalager](https://arctos.database.museum/agent/21324779)
   * etc etc as many of these as necessary by **agent** on **date** with **remarks**

probably with a unique key across (type, agent) - so many agents could assert the same thing for an identifier, or 1 agent could assert many things for one identifier, but one agent could not assert the same thing multiple times for the same identifier.

**Comment by Jegelewicz** [comment]
Got it, so these things would be "attributes of
![image](https://user-images.githubusercontent.com/5725767/165766808-0816a231-f1e4-45bb-b426-8e838c680815.png)

And probably we would organize them below the identifier just as we organize part attributes "under" parts?
![image](https://user-images.githubusercontent.com/5725767/165799210-2da73d36-f102-4996-a87e-2ead3e695f7e.png)

**Comment by Jegelewicz** [comment]
I think that is pretty much what everyone had in mind when it all seemed so confusing!

**Comment by dustymc** [comment]
Yes ``identifier_metadata`` (unless someone comes up with a better name) would be "attributes" of individual identifiers, in exactly the same way ``specimen_part_attribute`` are attributes of ``specimen_part``.

Organize is "just UI" but yes, similar structure so I'd expect similar interface.

So if that's all clear and acceptable, then the remaining question is what do ya'll want to be done which can't be done with the proposed structure?

And "details" I hope: 
* should the date component be DATE (day-precision native datatype) or ISO8601 (variable-precision 'carefully controlled string'), and 
* should it be required or optional?

**Comment by campmlc** [comment]
I agree

**Comment by dustymc** [comment]
>Yes

Which one?

>there isn't a date that is clearly correct

Yes, but it wasn't tomorrow or next year, and that might be important in a decade. I don't have any strong preferences either way, but that sort of thing does come up from time to time. (And it doesn't have to be lots of work, default now() could be implemented, but maybe that's pushing it.)

**Comment by Jegelewicz** [comment]
>>     Yes
> 
> Which one?

DOH! ISO8601 (variable-precision 'carefully controlled string')

**Comment by Jegelewicz** [comment]
> but it wasn't tomorrow or next year, and that might be important in a decade. I don't have any strong preferences either way, but that sort of thing does come up from time to time. (And it doesn't have to be lots of work, default now() could be implemented, but maybe that's pushing it.)

now() will probably be clearly wrong at least 30% of the time, but maybe that's no worse than nothing? Let others chime in!

**Comment by Jegelewicz** [comment]
> I think I heard "go" somewhere in the AWG meeting? Any objections, requests, changes, ??

I heard that too, but I think before it is implemented we need to make sure everyone is happy with #4605

**Comment by campmlc** [comment]
Example of ()?

**Comment by campmlc** [comment]
I heard go as well. Support implementation

**Comment by ewommack** [comment]
Definitely there was a go for collecting and recording the extra metadata. Suggestion was to put together mock-up suggestions for #4605 to share with AWG. So yes for the data, but not sure how it will look yet.

**Comment by campmlc** [comment]
per conversation with global interactions today - 
Can we get the ability to specify the assigned by and date as well as remarks in other ID metadata as soon as possible? 
We need to be able to assert that a relationships is asserted by GlobI, or at the very least, by an Arctos operator based on GloBi information. This is for SPNHC poster.

**Comment by Jegelewicz** [comment]
Also from that same meeting - Jorrit suggested that he would prefer to publish the relationships he finds and that we pick them up. All we need to do is provide a format for the publication (suggest the other identifier bulkload tool once we have the model solidified).

**Comment by Jegelewicz** [comment]
> use that as username in the otherID bulkloader

That means he needs to be an operator AND have permission for data_entry in ALL collections?

**Comment by Jegelewicz** [comment]
[Jorrit Poelen](https://arctos.database.museum/agents.cfm?agent_id=21339975)

**Comment by dustymc** [comment]
>needs to be an operator

Maybe - could be discussed, but I think it's semi-reasonable to limit that to Operators.

>AND have permission for data_entry in ALL collections?

Definitely not, there's some long discussion probably in the original component loader issue.

**Comment by dustymc** [comment]
See https://github.com/ArctosDB/arctos/issues/4722

If that flies, you'll probably want a 'bot' user to write the data. https://arctos.database.museum/agent/21339975 can be its parent (or whatever), and with that you can distinguish between the things Jorrit actually looked at vs. the things some script Jorrit wrote magicked out of the aether.

**Comment by campmlc** [comment]
I think we should move forward with this as unlikely we'll get more
feedback until it's implemented.

**Comment by dustymc** [comment]
Crud, the model as proposed can't accommodate #5135, I'm taking this back to needs discussion. Would be very very useful to round up any other nontheoretical use cases to make sure they fit before attempting this again. (@campmlc I think you had another recently but I can't find it?)

There some code in migration/__IDMetadata.sql.

This is a logical place for bot-fetched data from "the other end of the identifier" (host data in parasite records, for example), I think it makes sense to accommodate that.

People-asserted data - 'verbatim host ID' and such - would be a lot more accessible from here as well, but that can be a conversation for another day.

I believe the best way to accomplish that is some sort of typed/categorized/labeled free-text

* locality=some place-data pulled from another record
* identification=some ID-data pulled from another record

etc. That still needs accompanied by agent_id (to distinguish maybe-crazy things asserted by bots from 'data' asserted by humans).

I'm having a hard time reconciling that with the initial proposal. "issued" and "assigned" and such could go in the same list as locality and identification and such, but there's no "value" associated with those - Some Agent issued an ID and that's it. Just leaving the new field NULL in that case somehow hurts my brain - there'd be essentially two ways to see the data, depending on the 'type' (I think), and strange combinations would be possible - Some Agent issued locality data and such. (And maybe that's not a concern, this is fairly specialized and probably won't be used by eg new data entry students.)

The short version of the model as previously proposed:

* agent
* role
* date
* remark

New proposal that I don't think I like:


* blamable agent
* assertion type (code-table controlled, probably??)
* assertion value (must be free-text to accommodate eg hosts-of-parasites' locality data)
* date
* remark

examples:

* blamable agent ==> bot
* assertion type    ==> locality
* assertion value  ==> over there
* date      ==> whatever
* remark   ==> whatever

---------

* blamable agent ==> some organization
* assertion type    ==> issued identifier
* assertion value  ==> ??? 
* date      ==> whatever
* remark   ==> whatever


HELP!!

**Comment by Jegelewicz** [comment]
> the model as proposed can't accommodate https://github.com/ArctosDB/arctos/issues/5135,

I need an explanation of why - not sure this can be done via Github...especially since this seems to somehow exploded beyond other IDs somehow?

**Comment by dustymc** [comment]
I'm not sure if you're asking why the initally-proposed model isn't sufficient or why we need to accommodate those data?

I don't think there have been any explosions: identifiers establish relationships, caching some of the information from the other end of the relationship for both display and query makes various things easier (as in, brings into the realm of 'possible').

**Comment by campmlc** [comment]
I can look this over in detail later, but my recent example of needed metadata was KNRC: Koyukuk Nowitna Refuge Complex - where the Koyukuk Nowitna Refuge Complex should be the author of the ID. I think "issued by" may be confusing, as it could be interpreted as "who asserts this ID is applicable to this record", whereas "author" seems more clear?
In general, I support having agent names in metadata for ID author and ID assigned by. Also would be good to have ID remarks.

**Comment by Jegelewicz** [comment]
Maybe the things I said above should be attributes of the other identifier.

**Comment by dustymc** [comment]
>attributes

That works fine for prefetch

* attribute_type=whatever
* value=whatever
* (not sure units is ever necessary at all here??)
* determiner=some bot 

but I'm not sure how it works for 'issued' and such

* attribute_type=issuer (??)
* attribute_value= ??? The identifier?? 
* determiner=issuer, I suppose, since that's the agent and the agent is what issuer is trying to get at??

**Comment by Jegelewicz** [comment]
> but I'm not sure how it works for 'issued' and such

    attribute_type=issued by
    attribute_value= The agent who issued it (can we tie this to the agent table?)
    determiner= the person who determined the agent in value issued the identifier

**Comment by campmlc** [comment]
Bringing this back into discussion. From the perspective of a genomics collection, this information is critical. And it affects the UI upgrade. #5510 #5333

**Comment by dustymc** [comment]
@campmlc this hasn't fallen off of anybody's radar, I don't know how to proceed, I am asking for assistance. (And I'm being all picky about needing the kind of assistance that I can write code to - I still don't know how to defensibly use keys as attribute values, and I don't see any evidence that anyone else does either.) Please see above. 

I'm probably starting to sound like a broken record on this, but I'm doing so because it's been consistently working: This seems like a good candidate for a Task Force.

**Comment by campmlc** [comment]
I guess I don't understand what you need. A go ahead to proceed? Or if more than that, can you clarify? I don't understand what the technical limitations are, if any. Happy to set up a task force, but we need to know what we are tasked with doing.

**Comment by dustymc** [comment]
I don't think I can be more clear than I've been above.

The model that's been proposed cannot do the things that have been mentioned as expected of it. I don't know if that's because something has changed or because I haven't understood something. Task Force's first (possibly only) task would be to get us all on the same page, everyone knowing what this thing is expected to DO in sufficient details to build models and write code to them.

**Comment by Jegelewicz** [comment]
An interesting idea came about in our observation/media discussion.

The information we would like to add to other identifiers resembles the information we currently associate with media:

labels
relationships

For those identifiers that include a base url, what specifically makes them different from media?

Take this MSB:Host record - https://arctos-test.tacc.utexas.edu/guid/MSB:Host:10001

which has an associated parasite

![image](https://user-images.githubusercontent.com/5725767/221947305-254ff04d-7967-4966-92c7-d69759f8f622.png)

What we would like to do here is record the following information about [MSB:Para:6598](http://arctos.database.museum/guid/MSB:Para:6598):

- **issued by** [Museum of Southwestern Biology Parasite Collection](https://arctos-test.tacc.utexas.edu/agents.cfm?agent_id=21320869) according to Mariel L. Campbell on 2012-08-31 
- **is parasite of** MSB:Host:10001](http://arctos.database.museum/guid/MSB:Host:10001) according to Robert L. Rausch on xxxx-xx-xx

which with appropriate terms in the code tables and just a little tweaking (I think) we could do this in media because there is flexibility there to record information about linked items. Also, it seems like whenever we want to associate some url with a catalog record - the response is to use media.

I guess we were all wondering what makes this url

http://arctos.database.museum/guid/MSB:Para:6598

different from this one?

https://web.corral.tacc.utexas.edu/arctos/HWML/2022-10-05/HWML_153377_Lens3p2x_Optv1p25x_whole_body.JPG

When is a url "media", when is it not, and why?

**Comment by dustymc** [comment]
From Task Force Identifiers Awesomification: We have a plan, going next task, high (but not panicky-high) priority.

MODEL CHANGES

add:

* issued_by_agent_id
* remarks

-------------------

* ditch base_url - don't force anyone to deal in partial/ambiguous identifiers, don't need 500 categories for them (they're self-defining!), BUT
    * for entry (and editing and such) - allow users to pick a type and pull in the base URL and such
        *  recycle https://arctos.database.museum/info/ctDocumentation.cfm?table=ctcoll_other_id_type for this; worst possible functionality is same as current
        * maybe even make it smarter - make ``http://n2t.net/ark:/87602/m4/387108`` from ``http://n2t.net/ark:/87602/m4/387108`` and ``ark:/87602/m4/387108`` and ``87602/m4/387108`` and etc. (possible? maybe...)
* try to make sure IRI-like things get into the "good identifier" bucket, and no non-IRI-like things are allowed entry to the 'good identifier' bucket


--------------

* see how many collector/field/whatever we can merge
    * af, nk, etc. - "locally useful" - stay where they are with no changes
    * field number, collector number, preparator number, etc. - merge what's possible, suggest https://arctos.database.museum/info/ctDocumentation.cfm?table=ctcoll_other_id_type#other_identifier be used, see where we end up

**Comment by Jegelewicz** [comment]
MODEL CHANGES

add:

    issued_by_agent_id
    remarks

Is what we agreed to in the meeting. Once we have that, we can convert institutional identifiers WITHOUT a base url to a generic identifier type with the appropriate agent in issued_by_agent_id

I am working on ensuring that all appropriate agents exist so we can be ready to do this. My work is ongoing [here](https://docs.google.com/spreadsheets/d/1UGhcy_ATkIZQJXoU0GeKfL6VPK_84f1Ol9-ZQHKOEQ8/edit?usp=sharing).

**Comment by dustymc** [comment]
"Accession number" is a synonym for what we generally call catalog number for about half the users out there.

>to be able to tell the two apart

You can't do that from the terms alone, the instance you mention is likely already a duplicate, unless there's some actual evidence that these things are functionally different then I strongly suggest we collapse them into the same bucket. The values can retain the current type, nothing need be lost in this process.

**Comment by Jegelewicz** [comment]
The Park Service one may be a dupe, but this one isn't - https://arctos.database.museum/info/ctDocumentation.cfm?table=ctcoll_other_id_type#mvz_alternate_accession_number

I'd be happy to call all of these "institutional catalog number", but I think the users should weigh in.

**Comment by Jegelewicz** [comment]
> they are distinct (and the agent in the doc is just wrong, they're not centrally-assigned). 

I don't know how you know what the agent should be?

![image](https://user-images.githubusercontent.com/5725767/223220893-f44b64a8-ec6f-4cbe-bd13-b96654f06c88.png)

But if you do, then please let me know what agent I should be using if not https://arctos.database.museum/agents.cfm?agent_id=40

**Comment by dustymc** [comment]
> what the agent should be?

It's not that simple, some can be extracted from the IDs (I'm actually not sure even that is quite right), some probably can't. They need to stay as they are until there's an issue.

Those are both "catalog numbers" - but ALAAC (probably) needs to stay distinct for the same reasons NK does, it's used in the collection.

**Comment by Jegelewicz** [comment]
> It's not that simple, some can be extracted from the IDs (I'm actually not sure even that is quite right), some probably can't. They need to stay as they are until there's an issue.

But it is that simple now? 

U. S. National Park Service catalog

and 

institutional catalog number issued by U. S. National Park Service

without any change to the value in the identifier value field seems equivalent? If ids are determined to be from a specific park, the issuer can just be changed.

**Comment by dustymc** [comment]
This (the basics thereof) is in next release (in like an hour, maybe) and I want to close this issue (it's overly convoluted and wandered around a lot before arriving at a solution), but we're not done.

First, the immediately actionable: The just-above discussions strongly suggest we need a dedicated Issue (with user summoning and CSV) for each nonresolvable candidate for merging. That's going to be a lot of issues, but I think anything else is going to let things like ALAAC (which I just happen to know about because it influenced a bunch of Arctos development) slip through the cracks leading to (rightfully, ish) angry users and convoluted recovery and gnashing of teeth. A few hundred issues is way better - unless @Jegelewicz or @mkoo or someone wants to change my mind before ~tomorrow. (Please?! I'd so like a better idea about now! And I'm almost positive https://github.com/ArctosDB/internal/issues/251 would make that better so I'll probably use it if nobody saves or stops me, but what is the hangup?)

Second: The other stuff, magic around partial IDs and whatever - I don't think we can or should do anything immediately, babysteps, but there's a lot of room for improvement (in being unambiguous, and in making Arctos easier to use) so I don't want to completely forget about that either. I'll open more issues before flushing this one.

Other Second, which should maybe be first because timing: https://github.com/ArctosDB/arctos/issues/5193, this'll probably/maybe change bulkloader stuff, what (if anything) should I add to accommodate?

Did I miss anything else from above?

**Comment by campmlc** [comment]
Can the few hundred issues related to nonresolvable IDs be given a common
project name -e.g. "OtherID Metadata Resolution"? Either way it sounds
better to me to track these things as separate issues.
I support trying out the procedures - and if it doesn't work for some
reason, e.g. issues die and get buried waiting around for the 4th check, we
re-evaluate.
Do you need to finalize number of bulkloader columns immediately? Do we
need to do an emergency call tomorrow to resolve?

**Comment by dustymc** [comment]
I don't much use GH projects, but I can give them a common title-prefix-thingee so they should be easy to round up (and maybe add to projects if that helps someone).

> issues die and get buried

I would welcome a procedure regarding that, but someone (@ebraker ?) suggested framing data-changing issues as "will happen on date unless someone objects" and I think that's been working reasonably well. (Then things sometimes get stuck behind sometimes-not-very-solid objections for months, procedure - and/or better focus by the Issues Committee - is still needed, but it's better than it used to be.) That's what I'll plan for with these.

Help in reaching out to the collections would be very welcome/useful. Maybe we can find an easy one to use as an example or something - I'm guessing most aren't paying attention and have no idea what's going on, seeing real data tends to help.

> finalize number of bulkloader columns

Pretty much (and number includes structure, of course - I assume we want assigned in there but ????). I'm not sure a call is necessary, there are still hundreds of unused slots (that I've been trying to fill for near a half-year now), but I'm around if that's helpful.

This just went to production, go do something awesome with it!

**Comment by campmlc** [comment]
Edited first record with collector number: https://arctos.database.museum/guid/MSB:Mamm:135473. Yay!

This looks clear - but I think that for other records with just a numeric value for collector number (e.g. no initials first), it would be clearer to have the "issued by" between the ID type and the ID value, so that ID type = collector number, ID issued by =  collector agent name, and ID value = numeric value or alphanumeric value. Can we swap?

Also, as one thing always leads to another, now I'm so excited about this that I want to be able to find all records with my collector numbers and bulk update them with my agent name in issued by. Is this a tool that could be added to the Manage dropdown?

**Comment by campmlc** [comment]
Here's another one with more cryptic IDs explained. Debating whether its a good idea to add the MSB issued by for the NK number; from a data perspective, probably yes.
https://arctos.database.museum/guid/MSB:Mamm:222222

**Comment by dustymc** [comment]
>Can we swap?

I think so, but I'm not sure what you're asking for - clarify?

>bulk update

Certainly needs an Issue, probably has some interesting ways to be scary. And I'd like to think this is cleanup, so maybe start with some SQL updates? If you'll let me know how to find stuff I can update.

>good idea to add the MSB issued by for the NK 

Yea, I've been chewing on that for a while. Some of them are clearly "issued" by MSB, but those aren't the problems - it's the ones "issued" by some grad student who ran off with a pile of sheets (which might eventually find their way into Arctos), or "issued" by the stuck printer, or etc. (And there's surely a lost Nikita Khrushchev collector number in there somewhere.) Not a clue what the solution is, but it would be good to avoid further blurring that line/doing something that can't readily be un-done, which probably precludes a global update.

One more for the TODO list: othercatalognumbers is already a mess, don't think it's capable of carrying these, need to rebuild it as JSON and mitigate whatever that melts.

**Comment by campmlc** [comment]
I mean, move the "issued by" column to the left, maybe better on the right
side of the ID value, although just left of value works too.
So, in the case of the example I first posted:
https://arctos.database.museum/guid/MSB:Mamm:135473
the order would be:
Relationship (self): ID type (collector number): ID value: (MLC309), issued
by: (Mariel L Campbell), Link (blank), More, Containers, Extras, Assigned
by, Remarks

**Comment by dustymc** [comment]
Thx.

>although just left ... works too.

Oh - indeed! A fair bit of the utility in this is to merge some (demonstrably poorly used, check any of a dozen recent issues) types, so I've taken that one step further. The important thing about MLC309 is you - maybe it's correctly a collector number, maybe it ain't, maybe collector number will be around in a decade, maybe it won't, but with the link between you and the identifier that probably doesn't matter too much. So I've gone all in, that's now (correctly and spectacularly, I hope - because I've patched it to production) Mariel L. Campbell collector number MLC309. Yay everybody?

**Comment by Jegelewicz** [comment]
I spent three days working on [this](https://docs.google.com/spreadsheets/d/1UGhcy_ATkIZQJXoU0GeKfL6VPK_84f1Ol9-ZQHKOEQ8/edit?usp=sharing) - I can create issues one by one for each identifier and add them all to a project.

**Comment by dustymc** [comment]
Yep, I plan to use that. I don't think you have sufficient access to create the issues.

**Comment by Jegelewicz** [comment]
> I don't think you have sufficient access to create the issues.

?

You should not waste time creating all of these issues - I can create them and you can add whatever you need to add.

**Comment by Jegelewicz** [comment]
Test issue #5770 Let's make sure we have a template using this.

**Comment by dustymc** [comment]
We can try that. But, how'd you make the distinction between institutional catalog number and other identifier? Looks completely arbitrary to me - how's Albuquerque Biopark not an institution?! Suggest we converge on ``other identifier`` unless someone comes up with a nonarbitrary definition in very immediate future.

**Comment by Jegelewicz** [comment]
> how'd you make the distinction between institutional catalog number and other identifier

"museum" collections and everybody else - I don't really care what generic type gets used and each issue can determine what is best.

**Comment by dustymc** [comment]
Then please propose 'other identifier' for all.

**Comment by campmlc** [comment]
The ones I added last night were not institutional IDs, just my collector
numbers in variously entered random formats, and a couple of other cryptic
ids that referenced institutions related to the collector and event. I
think these types of situations may need to be resolved at the level of the
collection. But we should go with Teresa's list as the obvious first pass.

**Comment by dustymc** [comment]
>Other identifier in my understanding

I think that's about right, but it's no longer true. (Or at least I don't think it needs to be.)

The primary purpose of identifiers is links to data. That hasn't changed.

Where we're coming from, for nonresolvable identifiers, Albuquerque Biopark says "go talk to those folks" (and good luck, you'll need it for many of them).

Now, the "those folks" is an explicit agent reference. If you really know some exceptional specific use case (eg NKs are plastered all over everything at MSB and backed up by filed datasheets, and I suspect that's more than sufficient reason to keep them distinct - but who knows, maybe it'll turn out that a more-specific agent is a better tool even there) then maybe you need a distinct type. If all you REALLY know is "something issued by BLA" then you don't need type, you just check BLA's agent record (where you will hopefully find contact info, links to their data in archive collections, etc., etc.) and go ask them what 12345 means (or maybe figure it out from other activity, also linked from their profile). There's little use for most types in this system, they just add confusion rather than clarity.

>I'd like to go through the older IDs now with this info and clean them up.

There will be an issue with attached data for each.

And on that note, https://github.com/ArctosDB/arctos/issues/5770 and https://github.com/ArctosDB/arctos/issues/5771 look good to me - @Jegelewicz, would you like to open the floodgates? I'll try to keep up.

**Comment by Jegelewicz** [comment]
> would you like to open the floodgates?

Before I start - can we put up a short banner that links to #5771 and give everyone until next week to digest this? I'll start posting issues on Monday and just do as many as I can on any given day.

And I will actually post some today if I have time that are very collection specific and fairly straightforward. But I'm about to go into 3 hours of meetings, so it may be sparse...

**Comment by dustymc** [comment]
OK, I may post a few then, I see you've added what I was going to suggest to the spreadsheet so I'll keep track there. I'd like to get some things converted to folks can see how this works, and I'd like to make sure this is stable and usable before I go rip [everything](https://github.com/ArctosDB/arctos/issues/5331) apart for a some few-I-hope weeks. I'm not quite sure of - well anything I suppose, but getting some stuff rolling seems like a good start....

And whatever of this is necessary for the bulkloader still needs added to that issue.
